packages/@atjson/renderer-react/src/index.ts

```diff
@@ -73,20 +72,32 @@ export default class ReactRenderer extends Renderer {
   ): Iterator<void, ReactElement | ReactElement[], ReactElement[]> {
     this.renderSubdocuments(annotation);
 
-    let AnnotationComponent =
-      this.componentLookup[annotation.type] ||
-      this.componentLookup[classify(annotation.type)];
+    const annotationChildren = yield;
+    const key = `${annotation.id}-${annotation.start}`;
 
-    if (AnnotationComponent) {
-      return React.createElement(
-        AnnotationComponent,
-        annotation.attributes,
-        ...(yield)
-      );
-    } else {
-      // console.warn(`No component found for "${node.type}"- content will be yielded`);
-      let components = yield;
-      return components;
-    }
+    return React.createElement(ReactRendererConsumer, {
+      key,
+      children: (componentMap: ComponentMap) => {
```

Comment:
The type checker complained when I passed this function as the third argument (which should be the `children`), but is fine with this 🤷‍♂

Reply:
The type checker works in mysterious ways 🤷‍♂